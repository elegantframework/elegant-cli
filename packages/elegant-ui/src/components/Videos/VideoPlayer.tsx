import React from 'react';

interface Props {
  /**
   * The title of the video.
   */
  title: string;
  /**
   * The id of the video provided by YouTube. Ex. bS66QUBKljM
   */
  videoId: string;
  /**
   * The width of the video player.
   */
  width?: number | string;
  /**
   * The height of the video player.
   */
  height?: number;
  /**
   * Can the video player be viewed in fullscreen mode?
   */
  allowFullScreen?: boolean;
}

/**
 * A simple video player component for displaying videos from external websites.
 * @returns An iframe element containing a Youtube video player.
 */
const VideoPlayer = ({
  title,
  videoId,
  width = '100%',
  height = 350,
  allowFullScreen = true
}: Props) => {
  return (
    <>
        <iframe
          width={width}
          height={height}
          src={`https://www.youtube.com/embed/${videoId}`}
          title={title}
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          allowFullScreen={allowFullScreen}
        ></iframe>
    </>
  );
}

export default VideoPlayer;
