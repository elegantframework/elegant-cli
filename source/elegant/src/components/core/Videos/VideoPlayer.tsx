/**
 * 
 * @returns 
 */
const VideoPlayer = () => {
    return(
        <>
        
        </>
    );
};

export default VideoPlayer;