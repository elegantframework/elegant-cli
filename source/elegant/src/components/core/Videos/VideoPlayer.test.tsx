import React from "react";
import renderer from 'react-test-renderer';
import VideoPlayer from "./VideoPlayer";

describe("Video Player component", () => {
    it('renders a video player component properly', () => {
        const Player = renderer
        .create(
            <VideoPlayer 
                title="Hello World"
                description="This is a unit test!"
                embed_url="https://www.youtube.com/embed/bS66QUBKljM"
                content_url="https://youtu.be/bS66QUBKljM"
            />
        )
        .toJSON();
        expect(Player).toMatchSnapshot();
    });
});