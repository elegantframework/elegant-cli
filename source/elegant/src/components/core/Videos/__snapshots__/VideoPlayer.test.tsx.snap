// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Player component renders a video player component properly 1`] = `
<iframe
  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
  allowFullScreen={true}
  height={315}
  src="https://www.youtube.com/embed/bS66QUBKljM"
  title="Hello World"
  width={560}
/>
`;
